File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.java
class Solution {
    public int solution(int left, int right) {
        int ret = 0;
        for(int i = left; i <= right; i++){
            int n = 0;
            for(int j = 1; j <= i; j++){
                if(i % j == 0) n++;
            }
            if(n % 2 == 0) ret += i;
            else ret -= i;
        }
        return ret;
    }
}

File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.py
def iseven(num):
    cnt = 0
    for x in range(1, num+1):
        if x ** 2 > num:
            break
        if num % x == 0:
            if x ** 2 == num:
              cnt += 1
            else: cnt += 2
    return cnt % 2 == 0
        
def solution(left, right):
    return sum(x if iseven(x) else -x for x in range(left, right+1))